### PR TITLE
Bug 1791- 'Select currency' is displayed if we go to store and there is no currency from previous store.

### PR DESCRIPTION
**Pull request by AleksandrGogolushko**

### CODE REVIEW

File: src/app/component/CurrencySwitcher/CurrencySwitcher.component.js [hunk 1]
```diff
@@ -31,15 +31,28 @@ export class CurrencySwitcher extends PureComponent {
...
                 available_currencies_data: availableCurrencies,
                 current_currency_code: currentCurrencyCode
             } = {}
         } = this.props;
 
+        const currency = getCurrency();
+        const value = availableCurrencies.some((e) => e.id === currency) ? currency : currentCurrencyCode;
+
+        return value;
+    }
```
Comment (yeegor) [inline src/app/component/CurrencySwitcher/CurrencySwitcher.component.js:46]: Please comment on this, it is unclear how this should function and why so.

File: src/app/component/CurrencySwitcher/CurrencySwitcher.component.js [hunk 2]
```diff
@@ -31,15 +31,28 @@ export class CurrencySwitcher extends PureComponent {
         handleCurrencySelect: PropTypes.func.isRequired
     };
 
-    render() {
+    valueForSwitcher() {
```
Comment (yeegor) [inline src/app/component/CurrencySwitcher/CurrencySwitcher.component.js:34]: Please rename this function, e.g. to getCurrencyValue or something like that.

**Review by yeegor — CHANGES_REQUESTED** [review]

File: src/app/component/CurrencySwitcher/CurrencySwitcher.component.js (hunk 1 above)
Comment (AleksandrGogolushko) [reply src/app/component/CurrencySwitcher/CurrencySwitcher.component.js:46]: Save the value of the selected currency to the currency variable. Then  iterate over all the available currencies and if the selected currency matches the available ones, then we return the selected one, if not, then we return the default currency

Comment (yeegor) [reply src/app/component/CurrencySwitcher/CurrencySwitcher.component.js]: When I say that the logic is not obvious and ask you to comment on it, I imply that the next person who is going to work with this code (e.g. me) will not understand it in 10 seconds, and that means that this piece of code should be commented on directly in the code. Nobody will look for the information in the CRs :)

**Review by yeegor — CHANGES_REQUESTED** [review]

**Review by alfredsgenkins — APPROVED** [review]

**Pull request merged**